### PR TITLE
feat(signing): add user login endpoint

### PR DESCRIPTION
**Pull request by mekzy-o**
### What does this PR do?

- Creates functionality for users to signing

#### Description of Task to be completed?
- When a user goes to the landing page of author's haven, then the user should be able to login to his account
- User should be able to sign up via the route: `api/v1/auth/signin`
- Document login endpoint using swagger.json
- Write tests to validate controller

#### How should this be manually tested?
After cloning this repo, `cd` into it and do the following:

```bash
# fetch this branch
$ git fetch origin ft-user-signing-167165054

# install dependencies
$ npm install

# run unit tests
$ npm test

# start the development server
$ npm run server
```

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167165054](https://www.pivotaltracker.com/story/show/167165054)

#### Screenshots (if appropriate)

Screenshot 2019-07-28 at 5 52 31 PM

### CODE REVIEW

File: controllers/AuthController.js [hunk 1]
```diff
@@ -60,8 +60,57 @@ export default class AuthController {
...
+   * @static
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {object} returns user data
+   * @memberof AuthController
+   */
+  static async userSignin(req, res) {
+    const { email, password } = req.body;
+
+    const userEmail = await models.User.findOne({ where: { email } });
```
Comment (mazma1) [inline controllers/AuthController.js]: Looking at this variable name, does this request return a user's email?

File: __tests__/authRoutes.test.js [hunk 2]
```diff
@@ -44,6 +44,52 @@ describe('User signup tests', () => {
     });
   });
 });
+
+describe('User login tests', () => {
+  describe('test for user login', () => {
```
Comment (mazma1) [inline __tests__/authRoutes.test.js]: Why do you have a second `describe` block here?

File: controllers/AuthController.js [hunk 3]
```diff
@@ -60,8 +60,57 @@ export default class AuthController {
...
+      });
+    }
+    return res.status(200).send({
+      message: 'You have successfully logged in',
+      user: {
+        token,
+        email: userEmail.dataValues.email,
+        firstName: userEmail.dataValues.firstName,
+        lastName: userEmail.dataValues.lastName,
+        isVerified: userEmail.dataValues.isVerified
```
Comment (mazma1) [inline controllers/AuthController.js]: Please destructure  these values

File: controllers/AuthController.js [hunk 4]
```diff
@@ -43,13 +43,13 @@ export default class AuthController {
       isVerified: false,
       verificationToken,
       password: hashedPassword,
-      type: 'user',
```
Comment (tolulope-od) [inline controllers/AuthController.js]: Could you please add the trailing comma back?

File: controllers/AuthController.js (hunk 1 above)
Comment (mekzy-o) [reply controllers/AuthController.js]: the variable `email` is used to store the user's email in the request body...

Comment (mekzy-o) [reply controllers/AuthController.js]: is that what you meant?

File: controllers/AuthController.js (hunk 3 above)
Comment (mekzy-o) [reply controllers/AuthController.js]: Thanks. I would do that

File: controllers/AuthController.js (hunk 1 above)
Comment (mazma1) [reply controllers/AuthController.js]: I meant `userEmail`. Does the database request return a user's email?

Comment (mekzy-o) [reply controllers/AuthController.js]: Okay. thanks for pointing that out. would change the variable name. it actually returns  the 
user object

**Pull request merged**